sim/dex-species.ts

```diff
@@ -575,9 +575,17 @@ export class DexSpecies {
 	getMovePool(id: ID, isNatDex = false): Set<ID> {
 		let eggMovesOnly = false;
 		let maxGen = this.dex.gen;
+		const gen3HMMoves = ['cut', 'fly', 'surf', 'strength', 'flash', 'rocksmash', 'waterfall', 'dive'];
+		const gen4HMMoves = ['cut', 'fly', 'surf', 'strength', 'rocksmash', 'waterfall', 'rockclimb'];
 		const movePool = new Set<ID>();
 		for (const {species, learnset} of this.getFullLearnset(id)) {
 			for (const moveid in learnset) {
+				if (gen4HMMoves.includes(moveid) && this.dex.gen >= 5) {
+					if (!learnset[moveid].some(source => source.startsWith('5'))) continue;
+				} else if (gen3HMMoves.includes(moveid) && this.dex.gen >= 4 &&
+					!learnset[moveid].some(source => source.startsWith('4'))) {
+					continue;
+				}
```

Comment:
This fails for Hoothoot and Noctowl who learn Flash by HM in Gen 3 and by also by Virtual Console transfer from Gen 2 to Gen 7. (Although currently `/ms8` is still buggy, because it should show Flash being learnable in Gen 8.) Maybe it would be better to filter out sources that start `4` and `3` from the learnset?

Reply:
(Oh and there are also some CAP mons with weird learnsets where they learn HMs in Gen 4 and then again in Gen 7.)

Reply:
Flash is excluded from `/ms8` because it's a past gen only move. The code you highlighted would have failed to return Flash in `/ms7` but I added another commit that fixed this.

Reply:
Regarding the CAP mons, I'm not familiar with them but it seems like they're supposed to be exempted from HM transfer limitations? Krilowatt learns Surf in Gen 4 but not Gen 5, which in addition to causing this code to not return Surf in `/ms5 Krilowatt`, makes it show up as an illegal move in the Gen 5 CAP teambuilder, but passes validation.

I also noticed that Darkrai can have Rock Climb in Gen 8 Natdex (which also shows up as illegal in the teambuilder), which doesn't seem to be right either.